csv-extractor/csv_extractor.py

```diff
@@ -185,19 +203,20 @@ def process_files(client, monitor, paths, time_series_cache, failed_path) -> Non
                     path.replace(failed_path.joinpath(path.name))
 
             else:
-                logger.info("Processed {} datapoints from {}".format(count_data_points, path))
-                monitor.successfully_processed_files_gauge.inc()
-                monitor.count_of_time_series_gauge.set(count_ext_ids)
-                monitor.incr_total_data_points_counter(count_data_points)
-
-                path.unlink()
+                if queue.full():
+                    t = threading.Thread(target=_log_error, args=(
+                        post_all_data, queue, monitor, remaining_files_count))
+                    t.start()
+                    with queue.mutex:
+                        queue.queue.clear()
 
         except IOError as exc:
             logger.warning("Failed to delete/move file {}: {!s}".format(path, exc))
 
         remaining_files_count -= 1
         monitor.unprocessed_files_gauge.set(remaining_files_count)
         monitor.push()
+    print("toptal time taken...", time.time() - st)
```

Comment:
print statements aren't very useful in production code.
Maybe a log statement with some more details would be nice to have?